src/main/java/seedu/address/model/person/NextOfKin.java

```diff
@@ -31,7 +32,18 @@ public NextOfKin(String nextOfKin) {
      * Returns true if a given string is a valid Next of Kin.
      */
     public static boolean isValidNextOfKin(String test) {
-        return test.matches(VALIDATION_REGEX);
+        try {
+            String[] nextOfKinData = test.split(",", 3);
+            String name = nextOfKinData[0].trim();
+            String relationship = nextOfKinData[1].trim();
+            String contact = nextOfKinData[2].trim();
+            Boolean isValidName = !name.equals("") && name.matches(STRING_VALIDATION_REGEX);
+            Boolean isValidRelationship = !relationship.equals("") && relationship.matches(STRING_VALIDATION_REGEX);
+            Boolean isValidContact = !contact.equals("") && contact.matches(INTEGER_VALIDATION_REGEX);
+            return isValidContact && isValidRelationship && isValidName;
+        } catch (ArrayIndexOutOfBoundsException e) {
```

Comment:
Perhaps, you could add a check for if `nextOfKinData` is of size 3 rather than catching an `ArrayIndexOutofBoundsException` to test for missing inputs.